File: 02_Lua/AddTwoNums.lua
--Author: Pat Sipes
-- Add Two user numbers together

--variables
local num1 = nil
local num2 = nil
local sum = 0

-- Get first number 
while (num1 == nil or num1 == "" or not tonumber(num1)) do
    print("What is the first number: ")
    num1 = io.read()
end
--Get Second number 
while (num2 == nil or num2 == "" or not tonumber(num2)) do
    print("What is the second number: ")
    num2 = io.read()
end

--Add numbers together
sum = num1 + num2

-- Print total
print(num1.." + "..num2.." = "..sum)

File: 04_Python/Pseudocode/AddTwoNums.py
# Author: Pat Sipes
# Two nums from user
# Print sum
# print("Hello from Two Nums")

#if integer or float, return true
#else return false
def CheckIfNum(userInput):
    return bool(userInput.isdigit() or isinstance(userInput, float))


#keep asking for a number until valid number given
def TryAgain(userInput):
    while not CheckIfNum(userInput):
        userInput = input("That's not a valid number, try again: ")
    return userInput

num1 = input("Give me a number: ")
num1 = TryAgain(num1)

num2 = input("Give me another number: ")
num2 = TryAgain(num2)

print(num1 + " + " + num2 + " = " + (str((float(num1) + float(num2)))))

